iconfetcher.go

```diff
@@ -0,0 +1,108 @@
+package jujusvg
+
+import (
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+)
+
+// An IconFetcher provides functionality for retrieving icons for the charms
+// within a given bundle.  The FetchIcons function accepts a bundle, and
+// returns a dict mapping charm paths to icon data.
```

Comment:
No such thing as dicts in Go.

    // and returns a map from charm path to SVG icon data.

?

I'd suggest using a map[string][]byte which gives a better indication
of the contents of the map (and is a little more efficient too)